znvis/visualizer/headless_visualizer.py

```
@@ -0,0 +1,310 @@
+    ----------
+    particles : list[znvis.Particle]
+            A list of particle objects to add to the visualizer.
+    counter : int
+            Internally stored counter to track which configuration is currently
+            being viewed.
+    """
+
+    def __init__(
+        self,
+        particles: typing.List[znvis.Particle],
+        vector_field: typing.List[znvis.VectorField] = None,
```

Ruff (0.14.0)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)